Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
Thank you @rnystrom ! I'm not a big fan of how IGListKit forces sometimes that some protocols are meant to be implemented by certain classes, but I also acknowledge that in order to achieve such power and flexibility, some drawbacks will always be in the way. With that said, I didn't know the `isFirstSection` property which is exactly what I need for my specific problem. I'll keep having a separate object implementing the protocol, but I'll set this property in it. I'm closing both my PR and my issue. I'm sorry for wasting your time - at least this ended up being productive, right? 😄